app/src/main/java/communi/dog/aplicatiion/DB.java

```diff
@@ -176,7 +177,12 @@ public boolean isValidUserPassword(String userId, String userPassword) {
 
     public void updateUser(String userId, String userEmail, String userPassword, String userName, String phoneNumber, String dogName, String userDescription) {
         User newUser = new User(userId, userEmail, userPassword, userName, phoneNumber, dogName, userDescription);
-        this.usersRef.child(userId).setValue(newUser);
+        this.usersRef.child(userId).setValue(newUser).addOnSuccessListener(new OnSuccessListener<Void>() {
+            @Override
+            public void onSuccess(Void aVoid) {
+                currentUser = newUser;
```

Comment:
change the current user only if the id of the updated user is the same as the current one.
please add this check
I know that right now the only usage of this function is when changing the current user, but since someone else might use this function elsewhere in a different context it is good practice to keep it as general as possible